mmaction/core/runner/omnisource_runner.py

```diff
@@ -0,0 +1,162 @@
+    dataloader[0], which is the main dataloader.
+    """
+
+    def run_iter(self, data_batch, train_mode, source, **kwargs):
+        if self.batch_processor is not None:
+            outputs = self.batch_processor(
+                self.model, data_batch, train_mode=train_mode, **kwargs)
+        elif train_mode:
+            outputs = self.model.train_step(data_batch, self.optimizer,
+                                            **kwargs)
+        else:
+            outputs = self.model.val_step(data_batch, self.optimizer, **kwargs)
+        if not isinstance(outputs, dict):
+            raise TypeError('"batch_processor()" or "model.train_step()"'
+                            'and "model.val_step()" must return a dict')
+        # Since we have multiple sources, we add a suffix to log_var names,
+        # so that we can differentiate them.
+        if 'log_vars' in outputs:
+            log_vars = outputs['log_vars']
+            log_vars = {k + source: v for k, v in log_vars.items()}
+            self.log_buffer.update(log_vars, outputs['num_samples'])
+
+        self.outputs = outputs
+
+    def train(self, data_loaders, **kwargs):
+        self.model.train()
+        self.mode = 'train'
+        self.data_loaders = data_loaders
+        self.main_loader = self.data_loaders[0]
+        # Add aliasing
+        self.data_loader = self.main_loader
+        self.aux_loaders = self.data_loaders[1:]
+        self.aux_iters = [cycle(loader) for loader in self.aux_loaders]
+
+        auxiliary_iter_times = [1] * len(self.aux_loaders)
+        use_aux_per_niter = 1
+        if 'train_ratio' in kwargs:
+            train_ratio = kwargs.pop('train_ratio')
+            use_aux_per_niter = train_ratio[0]
+            auxiliary_iter_times = train_ratio[1:]
+
+        self._max_iters = self._max_epochs * len(self.main_loader)
+
+        self.call_hook('before_train_epoch')
+        time.sleep(2)  # Prevent possible deadlock during epoch transition
+
+        for i, data_batch in enumerate(self.main_loader):
+            self._inner_iter = i
+            self.call_hook('before_train_iter')
+            self.run_iter(data_batch, train_mode=True, source='')
+            self.call_hook('after_train_iter')
+
+            if self._iter % use_aux_per_niter != 0:
+                self._iter += 1
+                continue
+
+            for idx, n_times in enumerate(auxiliary_iter_times):
+                for step in range(n_times):
+                    data_batch = next(self.aux_iters[idx])
+                    self.call_hook('before_train_iter')
+                    self.run_iter(
+                        data_batch, train_mode=True, source=f'/aux{idx}')
+                    self.call_hook('after_train_iter')
+            self._iter += 1
+
+        self.call_hook('after_train_epoch')
+        self._epoch += 1
+
+    # Now that we use validate hook, not implement this func to save efforts.
```

Comment:
However, it was not used in any config of action recognition models. If users want to check the validation accuracy or loss, they are recommended to use the `validate` hook rather than `val pipeline`.